sparsecoding/inference.py

```diff
@@ -347,15 +376,16 @@ def infer(self, data, dictionary):
             
             # Set coefficients to zero before doing repeating the inference with new lambdas
             coefficients=torch.zeros(batch_size, n_basis, requires_grad=True)
```

Comment:
These coefficients should be cast to to dictionary device defined above.

Reply:
Yeap, updated